### PR TITLE
EVG-2746 Fix Github Hook Management Logic

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: migrations/github_hook.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			Repo   string `bson:"repo"`
+		}{}
+		err = query.One(&existingHook)
+		if err != nil && !db.ResultsNotFound(err) {
+			return err
+		}
+
+		if existingHook.HookID != 0 {
+			grip.Infof("Hook for %s/%s already exists as Hook #%d. Please delete %d",
+				existingHook.Owner, existingHook.Repo, existingHook.HookID, hookID)
```
Comment (tychoish) [inline migrations/github_hook.go]: message fields.

**Comment by richardsamuels** [comment]
Running a patch manually: https://evergreen.mongodb.com/patch/5a8201062fbabe26bcb2a064

File: migrations/github_hook.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+			return errors.New("Missing owner or repo")
+		}
+
+		// check if there is an existing github hook for the same repo
+		query = session.DB(database).C(githubHooksCollection).Find(bson.M{
+			ownerKey: ref.Owner,
+			repoKey:  ref.Repo,
+		})
+		existingHook := struct {
+			HookID int    `bson:"hook_id"`
```
[the remark this answers is not in the document]
Comment (richardsamuels) [reply migrations/github_hook.go]: Github provides no guarantee that hooks are globally unique.

[the remark this answers is not in the document]
Comment (richardsamuels) [reply migrations/github_hook.go]: And I already noted an index in this change, see scripts/indexes.js

[the remark this answers is not in the document]
Comment (tychoish) [reply migrations/github_hook.go]: sweet

**Comment by tychoish** [comment]
I think this is fine, but we should hold off to push until after the next deploy, to avoid a messy deploy situation.

**Pull request merged**